specs/openapi.yml

```diff
@@ -0,0 +1,199 @@
+openapi: 3.0.3
+info:
+  title: Anoma
+  description: Interacting with an Anoma blockchain via Tendermint RPC
+  version: 0.6.1
+servers:
+  - url: http://127.0.0.1:26657
+    description: Tendermint RPC endpoint for an Anoma ledger
+paths:
+  /:
+    post:
+      summary: Interact with the Anoma blockchain via Tendermint RPC
+      operationId: abci_query
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                id:
+                  description: Should be unique between requests
+                  type: integer
+                  example: 58392
+                method:
+                  description: The Tendermint RPC method being called which in this case should always be abci_query
+                  type: string
+                  enum:
+                    - "abci_query"
+                params:
+                  type: object
+                  required:
+                    - path
+                  properties:
+                    path:
+                      description: Path as will be recognized by the ledger
+                      oneOf:
+                        - type: string
+                          enum:
+                            - "epoch"
+                            - "dry_run_tx"
+                          description: >
+                            * `epoch` - Epoch at the given block height
+                            * `dry_run_tx` - Dry run a transaction
+                        - type: string
+                          description: Read a storage value with exact storage key
+                          pattern: r"^value\/([\x00-\x7F]|\/)+$"
```

Comment:
I'm not sure what pattern to allow here for a valid storage key, `[\x00-\x7F]` covers ASCII bytes (0-127), so this pattern can accept things like `value/#atest1v4ehgw36g4pyg3j9x3qnjd3cxgmyz3fk8qcrys3hxdp5xwfnx3zyxsj9xgunxsfjg5u5xvzyzrrqtn/public_key` with a `#` character in it, but maybe this is too wide (or too narrow?).

Reply:
currently we allow utf-8

Reply:
I've changed these regexes to permit any pattern for storage key (I guess it's too permissive as it would allow non-UTF-8 string).